### DISCUSSION TITLE
QR codes instead of barcode markers

### DISCUSSION THREAD

**Discussion by Ales-ingeteam** [AR-js-org/AR.js#240] [Q&A] [answered]
Hi! I´m new here, sorry if this goes on Issues instead of Discussions.
Is there any chance of using QR codes instead of barcode markers to show 3D objects with ar.js and a-frame?

I need to track and display a large amount of different 3D objects using markers and show data in form of text when you click on each object. I already show that data in pdfs on my web using qr codes so i would like to use that same qr codes to show it on ar.
Thanks in advance!

**Comment by jonnypage** [comment, marked as answer]
You could try making the QR codes NFT markers and see how that performs

**Reply by Ales-ingeteam** [reply]
Thanks! I´ll try that.

**Reply by Ales-ingeteam** [reply]
I´ve tried it and it doesn´t seem to work. I don´t get any errors but when I point to the QR nothing happens.
I used the NFT marker creator from this link https://carnaux.github.io/NFT-Marker-Creator/ and my route to the image folder is working fine.

**Reply by jonnypage** [reply]
yeah that's the correct generator, could be your QR code, can you post one of them?

**Reply by Ales-ingeteam** [reply]
Yes, ofc! Sorry for the delay. Here you have one:
![E8Vku7RPiCCh69uAogSmEcrEoMjRjIQb7kr3znngkzixNFAdlWtC3qdBdvRwklSW](https://user-images.githubusercontent.com/82440244/115201607-d079b180-a0f5-11eb-8954-37df74c533b4.png)

**Reply by Ales-ingeteam** [reply]
I have found the error, the object was to small to see it (silly me). Now the only thing that bothers me is that the displayed object moves a lot.

**Reply by jonnypage** [reply]
Glad you got it to work! 

Yeah, NFT tracking is not as precise as marker tracking (yet)

**Reply by Ales-ingeteam** [reply]
Thanks for all your help!
By any chance do you know if I it it works well with a marker inside of a QR? Or the best solution is to stop using QRs and use only barcode markers?

**Reply by jonnypage** [reply]
I've used barcode markers inside qr codes with great success, I created the markers using this site: https://www.qrcode-monkey.com - I just set the barcode marker to be the logo image, and the qr code to the url of the AR experience

**Reply by kalwalt** [reply]
Hi to all! Nice that you use the discussions! In our [documentation page](https://ar-js-org.github.io/AR.js-Docs/marker-based/) you can find a link to a medium article (thanks to @nicolocarpignoli!) to embed a marker into a QRcode. i think this could be helpful for you. :slightly_smiling_face:

**Reply by Ales-ingeteam** [reply]
I have already tried it following the documentation and creating first the custom qr with barcode using [https://www.qrcode-monkey.com/](url) and then uploading to [https://jeromeetienne.github.io/AR.js/three.js/examples/marker-training/examples/generator.html](url) to get the .patt file and it seems to work, but as with the NFT tracking the object moves a lot and disappears sometimes. I tried changing the smooth attributes too but it doesn´t improve the tracking.

I don´t know if I´m doing something wrong. This is my qr code with the marker in png format: 
![pattern-qr-code](https://user-images.githubusercontent.com/82440244/115837346-c3700180-a418-11eb-9921-f813eff7e026.png)

**Reply by kalwalt** [reply]
The NFT tracking is an experimental feature, one of the issue Is the jttering because the data is not perfectly filtered. We haven't found yet a better solution. If you are interested on this topic you can read the PR regarding NFT.  Back to the topic of your question: for sure Barcode will have better stability against NFT.

**Reply by Ales-ingeteam** [reply]
I´ll stick to barcodes then. Thank you both for all your help! :smile:
